utils/include/detray/detectors/create_wire_chamber.hpp

```diff
@@ -0,0 +1,231 @@
+
+            // Surfaces with the linking into the local containers
+            mask_link_type mask_link = {wire_id,
+                                        masks.template size<wire_id>()};
+            material_link_type material_link{rod_id,
+                                             materials.template size<rod_id>()};
+            const auto trf_index = transforms.size(ctx0);
+            surfaces.emplace_back(trf_index, mask_link, material_link,
+                                  volume_idx, dindex_invalid,
+                                  surface_id::e_sensitive);
+
+            // The wire bounds
+            masks.template emplace_back<wire_id>(
+                empty_context{}, mask_volume_link, cell_size, cyl_half_z);
+            materials.template emplace_back<rod_id>(empty_context{}, wire_mat,
+                                                    wire_rad);
+
+            // Build the transform
+            vector3 z_axis{0.f, 0.f, 1.f};
+            vector3 r_axis = vector::normalize(m_center);
+            const scalar sign = (i_lay % 2 == 0) ? 1 : -1;
+            z_axis = axis_rotation<transform3_t>(r_axis,
+                                                 sign * stereo_angle)(z_axis);
+            vector3 x_axis =
+                unit_vectors<vector3>().make_curvilinear_unit_u(z_axis);
+            transforms.emplace_back(ctx0, m_center, z_axis, x_axis);
+        }
+
+        // Iterate the surfaces and update their links
+        const auto trf_offset{det.transform_store().size(ctx0)};
+        auto sf_offset{det.n_surfaces()};
+
+        for (auto &sf_desc : surfaces) {
+            // Make sure the volume was constructed correctly
+            assert(sf_desc.volume() < det.volumes().size());
+
+            // Update the surface links accroding to number of data in detector
+            const auto sf = surface{det, sf_desc};
+            sf.template visit_mask<detail::mask_index_update>(sf_desc);
+            sf.template visit_material<detail::material_index_update>(sf_desc);
+            sf_desc.update_transform(trf_offset);
+            sf_desc.set_index(sf_offset++);
+
+            // Copy surface descriptor into global lookup
+            det.add_surface_to_lookup(sf_desc);
+        }
+
+        //
+        // Fill Grid
+        //
+
+        // Get relevant ids
+        using geo_obj_ids = typename detector_t::geo_obj_ids;
+        constexpr auto cyl_id = detector_t::masks::id::e_portal_cylinder2;
+        constexpr auto grid_id = detector_t::sf_finders::id::e_cylinder2_grid;
+
+        using cyl_grid_t =
+            typename detector_t::surface_container::template get_type<grid_id>;
+        auto gbuilder =
+            grid_builder<detector_t, cyl_grid_t, detray::detail::fill_by_pos>{};
+
+        // The disc portals are at the end of the portal range by construction
+        auto portal_mask_idx = (det.portals(vol).end() - 3u)->mask().index();
+        const auto &cyl_mask =
```

Comment:
Just fyi, this maps the grid on either the inner or outer cylinder portal, which is not an exact match with where the surfaces are. This is on my TODO list to fix in the toy detector. If you print the grid, you will e.g. see that multiple surfaces are mapped into a single bin etc